tests/test_redis_lock.py

```diff
@@ -499,3 +500,15 @@ def workerfn(unblocked):
     worker.join(0.5)
 
     assert unblocked.value == 1
+
+
+def test_auto_renewal_stops_after_gc(conn):
+    """Auto renewal stops after lock is garbage collected."""
+    lock = Lock(conn, 'spam', auto_renewal=True, expire=1)
+    name = lock._name
+    lock.acquire()
+    del lock
+    gc.collect()
+
+    time.sleep(2)
+    assert conn.get(name) is None
```

Comment:
In this test I would check internals - check if `thread.isAlive()` for a second or two. We really need to have a way to assert the thread is long dead. Maybe even a weakref on the tread? (to make sure the thread is gc-ed too)

Not sure about `gc.collect()` on PyPy ... I guess test results will tell if it really "collects".